ngraph/test/type_prop/einsum.cpp

```diff
@@ -0,0 +1,291 @@
+    auto O = make_shared<op::v7::Einsum>(OutputVector{I1, I2}, equation);
+    ASSERT_EQ(O->get_element_type(), element::f32);
+    ASSERT_TRUE(O->get_output_partial_shape(0).same_scheme(out_shape));
+}
+
+TEST(type_prop, einsum_dynamicshape_diagextraction)
+{
+    std::string equation = "xyzxy->xyz";
+    const auto input1_shape = PartialShape{Dimension(2, 7), Dimension(1, 5), 4, Dimension(3, 5), 3};
+    const auto out_shape = PartialShape{Dimension(3, 5), 3, 4};
+    auto I1 = make_shared<op::Parameter>(element::f32, input1_shape);
+    auto O = make_shared<op::v7::Einsum>(OutputVector{I1}, equation);
+    ASSERT_EQ(O->get_element_type(), element::f32);
+    ASSERT_TRUE(O->get_output_partial_shape(0).same_scheme(out_shape));
+}
+
+TEST(type_prop, DISABLED_einsum_dynamicshape_ellipsis1)
+{
+    // TODO: fix bug #53518 - PartialShape::broadcast_merge_into or Dimension::broadcast_merge
+    //  to support broadcasting between Dimension(3, 5) and Dimension(1, 3)
+    //  for which the result must be Dimension(3, 5)
+    std::string equation = "a...b,b...->a...";
+    const auto input1_shape = PartialShape{11, 1, Dimension(3, 5), 3};
+    const auto input2_shape = PartialShape{3, 11, 7, Dimension(1, 3)};
+    const auto out_shape = PartialShape{11, 11, 7, Dimension(3, 5)};
+    auto I1 = make_shared<op::Parameter>(element::f32, input1_shape);
+    auto I2 = make_shared<op::Parameter>(element::f32, input2_shape);
+    auto O = make_shared<op::v7::Einsum>(OutputVector{I1, I2}, equation);
+    ASSERT_EQ(O->get_element_type(), element::f32);
+    ASSERT_TRUE(O->get_output_partial_shape(0).same_scheme(out_shape));
+}
+
+TEST(type_prop, einsum_implicitmode_mixedcaseletters)
+{
+    // the following equation is equivalent to "AbC->ACb"
+    std::string equation = "AbC";
+    const auto input1_shape = PartialShape{1, Dimension(2, 3), Dimension(4, 5)};
+    auto I1 = make_shared<op::Parameter>(element::f32, input1_shape);
+    const auto out_shape = PartialShape{1, Dimension(4, 5), Dimension(2, 3)};
+    auto O = make_shared<op::v7::Einsum>(OutputVector{I1}, equation);
+    ASSERT_EQ(O->get_element_type(), element::f32);
+    ASSERT_TRUE(O->get_output_partial_shape(0).same_scheme(out_shape));
+}
+
+TEST(type_prop, einsum_implicitmode_mixedcaseletters2)
+{
+    // the following equation is equivalent to "a...b,B...->...Bab"
+    std::string equation = "a...b,B...";
+    const auto input1_shape = PartialShape{Dimension(3, 5), 11, 1, 3};
+    const auto input2_shape = PartialShape{Dimension(1, 3), 3, 1, 7};
+    const auto out_shape = PartialShape{3, 11, 7, Dimension(1, 3), Dimension(3, 5), 3};
+    auto I1 = make_shared<op::Parameter>(element::f32, input1_shape);
+    auto I2 = make_shared<op::Parameter>(element::f32, input2_shape);
+    auto O = make_shared<op::v7::Einsum>(OutputVector{I1, I2}, equation);
+    ASSERT_EQ(O->get_element_type(), element::f32);
+    ASSERT_TRUE(O->get_output_partial_shape(0).same_scheme(out_shape));
+}
+
```

Comment:
Add two tests where: one (all) of the input tensor(s) have dynamic rank.

Reply:
Added.